### PR TITLE
Add encryption to volume type

### PR DESCRIPTION
**Pull request by jsaroja66**

### CODE REVIEW

**Comment by manuelmazzuola** [comment]
hi @jsaroja66, please reference the PR to an issue.

File: core-test/src/main/java/org/openstack4j/api/network/TrunkTests.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import org.openstack4j.api.Builders;
+import org.openstack4j.model.common.ActionResponse;
+import org.openstack4j.model.network.Port;
+import org.openstack4j.model.network.SubPort;
+import org.openstack4j.model.network.Trunk;
+import org.openstack4j.openstack.networking.domain.NeutronSubPort;
+import org.testng.annotations.Test;
+
+/**
+ * Rewrote the entire API and so had to re-write tests
```
Comment (manuelmazzuola) [inline core-test/src/main/java/org/openstack4j/api/network/TrunkTests.java]: ok but no need to add the comment

File: core/src/main/java/org/openstack4j/api/networking/NetworkingService.java [hunk 2]
```diff
@@ -56,14 +56,14 @@
...
+
     /**
      * @return the LBaaS service
      */
     LoadBalancerService loadbalancers();
-    
+
     /**
-     * 
+     *
```
Comment (manuelmazzuola) [inline core/src/main/java/org/openstack4j/api/networking/NetworkingService.java]: do not reformat

File: core/src/main/java/org/openstack4j/api/networking/NetworkingService.java [hunk 3]
```diff
@@ -74,12 +74,15 @@
...
+     * <p>
+     * OpenStack Firewall As a Service <code>(FwaaS) : Firewall</code> Operations
+     * API
+     * </p>
+     *
      * @return the FwaaS service
      */
     FirewallAsService firewalls();
-    
+
```
Comment (manuelmazzuola) [inline core/src/main/java/org/openstack4j/api/networking/NetworkingService.java]: do not reformat

File: core/src/main/java/org/openstack4j/openstack/networking/builder/NeutronBuilders.java [hunk 4]
```diff
@@ -141,52 +146,52 @@ public HealthMonitorAssociateBuilder lbPoolAssociateHealthMonitor() {
...
 
     @Override
-    public ListenerV2Builder listenerV2(){
+    public ListenerV2Builder listenerV2() {
         return NeutronListenerV2.builder();
     }
 
     @Override
-    public ListenerV2UpdateBuilder listenerV2Update(){
+    public ListenerV2UpdateBuilder listenerV2Update() {
```
Comment (manuelmazzuola) [inline core/src/main/java/org/openstack4j/openstack/networking/builder/NeutronBuilders.java]: we will reformat in a separate pr

File: core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronTrunkUpdate.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    private String name;
+
+    @JsonProperty("admin_state_up")
+    private boolean adminStateUp;
+
+    @JsonProperty("description")
+    private String description;
+
+    public static NeutronTrunkUpdate update(Trunk t) {
+        NeutronTrunkUpdate u = new NeutronTrunkUpdate();
```
Comment (manuelmazzuola) [inline core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronTrunkUpdate.java]:
```suggestion
        NeutronTrunkUpdate neutronTrunkUpdate = new NeutronTrunkUpdate();
```

File: core/src/main/java/org/openstack4j/openstack/storage/block/domain/CinderEncryptionType.java [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this).omitNullValues().add("id", id).add("volumeTypeId", volumeTypeId)
+                .add("provider", provider).add("cipher", cipher).add("keySize", keySize)
+                .add("controlLocation", controlLocation).toString();
+    }
+
+    public static class ConcreteEncryptionTypeBuilder implements EncryptionTypeBuilder {
+
+        private CinderEncryptionType m;
```
Comment (manuelmazzuola) [inline core/src/main/java/org/openstack4j/openstack/storage/block/domain/CinderEncryptionType.java]:
```suggestion
        private CinderEncryptionType cinderEncryptionType;
```

File: core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockVolumeServiceImpl.java [hunk 7]
```diff
@@ -159,6 +168,27 @@ public VolumeType createVolumeType(VolumeType volumeType) {
...
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EncryptionType createEncryptionType(EncryptionType encryptionType, String volumeTypeId) {
+        checkNotNull(encryptionType);
+        checkNotNull(volumeTypeId);
+        return post(CinderEncryptionType.class, uri("/types/%s/encryption", volumeTypeId)).entity(encryptionType)
```
Comment (manuelmazzuola) [inline core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockVolumeServiceImpl.java]: new line .entity

**Comment by jsaroja66** [comment]
This pull request has 3 commits, I am going to make separate pull request for each of these commits.

**Comment by jsaroja66** [comment]
Will consider your comments in the new pull requests.

**Pull request closed without merging**